phe/util.py

```diff
@@ -427,3 +429,68 @@ def is_prime(n, mr_rounds=25):
             return False
     # the actual generic test; give a false prime with probability 2⁻⁵⁰
     return miller_rabin(n, mr_rounds)
+
+
+def int_div_round(a, b):
+    """Integer division with correct rounding.
+
+    The result is always an integer. Tiebreaking towards even. (Just
+    like float.)
+    """
+    assert isinstance(a, int)
+    assert isinstance(b, int)
+    if b & 1:
+        # b is odd. We don't have to worry about tiebreaking.
+        return (a + (b >> 1)) // b
+    else:
+        div, mod = divmod(a, b)
+        b_div_2 = b >> 1
+        # Round to nearest integer, away from even.
+        if mod > b_div_2 or (div & 1 and mod == b_div_2):
+            return div + 1
+        else:
+            return div
+
+
+def round_m_mul_2_pow_e_mul_b_pow_c(m, e, b, c):
+    """Compute round(m * 2 ** e * b ** e) for m, e, b, c integers.
```

Comment:
Typo in the equation (`e` instead of `c`).

Would be good to add a few regression tests checking that this function equals `round(m * 2 ** e * b ** e)` where it should...